### ISSUE TITLE
[Bluetooth] BluetoothUtils.toIntArray throwing type mismatch error from [Bluetooth.Airthings] (introduced in commit #13967) 

### ISSUE THREAD

**Issue by dw-8** [openhab/openhab-addons#15313]
## Current Behavior
With the recent changes made in commit #13967, the call to updateChannels(BluetoothUtils.toIntArray(data)) within org.openhab.binding.bluetooth.airthings.internal.AbstractAirthingsHandler is now silently failing for an Airthings Wave Gen 1.

This call is not converting the Byte data to Int data so sensor data is not getting updated.

Adding additional logging returned the following error message:
```
java.lang.ArrayStoreException: arraycopy: type mismatch: can not copy byte[] into int[]
	at java.lang.System.arraycopy(Native Method) ~[?:?]
```

## Expected Behavior
This call converts the Byte data to Int data and updates the sensor data

**New code (commit #13967)**
```
    public static int[] toIntArray(byte[] value) {
        int[] ret = new int[value.length];
        System.arraycopy(value, 0, ret, 0, value.length);
        return ret;
    }
```

**Previous code**
```
    public static int[] toIntArray(byte[] value) {
        if (value == null) {
            return null;
        }
        int[] ret = new int[value.length];
        for (int i = 0; i < value.length; i++) {
            ret[i] = value[i];
        }
        return ret;
    }
```

Reverting this commit back to the original code restores the functionality for Airthings Wave Gen 1.

## Your Environment
Debian 10 amd64
openHab 4.0
Zulu Embedded Java 17

**Comment by jlaur** [comment]
@dw-8 - thanks for your issue and detailed analysis. With that analysis I was able to quickly provide a pull request for reverting to the last known working state of the code. We aim to get this into a 4.0.1 hotfix release.

**Comment by jlaur** [comment]
@dw-8 - can you test [org.openhab.binding.bluetooth-4.1.0-SNAPSHOT.jar](https://drive.google.com/file/d/1wqW8EaxAmE6SwzjwM9idITA3mZccftj_/view?usp=sharing) just to be sure everything is working as expected?

**Comment by dw-8** [comment]
> @dw-8 - can you test [org.openhab.binding.bluetooth-4.1.0-SNAPSHOT.jar](https://drive.google.com/file/d/1wqW8EaxAmE6SwzjwM9idITA3mZccftj_/view?usp=sharing) just to be sure everything is working as expected?

@jlaur - After deleting the original bluetooth binding, dropping in your jar file, and restarting OH, it is now working as expected.

**Comment by jlaur** [comment]
@jlaur - After deleting the original bluetooth binding, dropping in your jar file, and restarting OH, it is now working as expected.

Thanks! Now you can revert that and upgrade to 4.0.1. 😉

**Comment by dw-8** [comment]
Re-opening this bug as it looks like @lsiepel reverted this bug fix inside #15828
